modules/vstudio/vs2005_dotnetbase.lua

```diff
@@ -820,8 +816,49 @@
 		end
 	end
 
+	function dotnetbase.netcore.getsdk(cfg)
+        local map = {
+			["Default"] = "Microsoft.NET.Sdk",
+            ["Web"] = "Microsoft.NET.Sdk.Web",
+            ["Razor"] = "Microsoft.NET.Sdk.Razor",
+            ["Worker"] = "Microsoft.NET.Sdk.Worker",
+            ["Blazor"] = "Microsoft.NET.Sdk.BlazorWebAssembly",
+            ["WindowsDesktop"] = "Microsoft.NET.Sdk.WindowsDesktop",
+            ["MSTest"] = "MSTest.Sdk",
+        }
+
+		if cfg.flags.WPF then
+			return map["WindowsDesktop"]
+        end
+
+        return map[cfg.dotnetsdk or "Default"]
+	end
+
+	function dotnetbase.netcore.dotnetsdk(cfg)
+		dotnetbase.generate_global_json(cfg)
+	end
+
 	function dotnetbase.allowUnsafeBlocks(cfg)
 		if cfg.clr == "Unsafe" then
 			_p(2,'<AllowUnsafeBlocks>true</AllowUnsafeBlocks>')
 		end
 	end
+
+	function dotnetbase.output_global_json(prj) -- unsure how you handle project with different config
+			if prj.dotnetsdk == "MSTest" then
+				_p([[
+{
+	"msbuild-sdks":
+		{
+			"MSTest.Sdk": "3.6.1"
+		}
+}
+		]])
+		end
+	end
+	function dotnetbase.generate_global_json(prj)
+				local content = p.capture(function() dotnetbase.output_global_json(prj) end)
+				if content ~= nil and #content > 0 then
+					p.generate(prj, path.join(prj.workspace.location,"global.json"), function() p.outln(content) end)
```

Comment:
```suggestion
					p.generate(prj, path.join(prj.workspace.location, "global.json"), function() p.outln(content) end)
```
BTW, I removed your check to not overwrite existing file for simplicity in precedent suggestion.
Unsure of your expected behavior with existing file.
If you keep existing code, doc should be updated as existing file would be overwritten.

Reply:
preferebly you would just append it to the global.json file or update it but have no idea on how to acomplish that with existing json api's

Reply:
gonna have a look at json.decode and encode to make it so it just updates the version or generates it when missing!